### PR TITLE
Setup Test Framework

### PR DESCRIPTION
**Pull request by quiqua**
This PR will add [Spek](http://spekframework.org/) as the testing framework.

To make this work, several things are changed/added:

- Upgrade gradle to 4.6+ (4.7 actually as it is the latest)
- Integrate plugin `de.mannodermaus.android-junit5` to enable JUnit5 with Android (JUnit5 is required for Spek) also since Gradle 4.6 there is a built-in support for JUnit Platform: https://docs.gradle.org/4.6/release-notes.html
- Added [Hamkrest](https://github.com/npryce/hamkrest) for assertions
- Added [MockK](https://github.com/mockk/mockk) for mocking (but no test uses this so far)

### CODE REVIEW

File: app/src/test/kotlin/eu/quiqua/gloomhaven/statstracker/features/stats/viewModel/StatsViewModelSpek.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        given("with a StatsViewModel") {
+            val statsViewModel = StatsViewModel(application)
+            on("getStats") {
+                it("should return the defined model with its stats") {
+                    assert.that(statsViewModel.getStats(), equalTo("hello world"))
+                }
+            }
+        }
+    }
+})
```
Comment (quiqua) [inline app/src/test/kotlin/eu/quiqua/gloomhaven/statstracker/features/stats/viewModel/StatsViewModelSpek.kt]: whops, newline missing :/

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -41,9 +41,26 @@ dependencies {
...
+        exclude group: "org.junit.platform"
+        exclude group: "org.jetbrains.kotlin"
+    }
+
+    testImplementation "io.mockk:mockk:$mockk_version"
+    // apply android-junit5 shenanigans
+    testImplementation junit5.unitTests()
+    testCompileOnly junit5.unitTestsRuntime()
+
+    testImplementation "com.natpryce:hamkrest:1.4.2.2"
```
Comment (quiqua) [inline gradle/dependencies.gradle]: I should use the version defined above here.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Thu May 17 15:06:52 CEST 2018
+#Fri Jun 01 15:41:02 CEST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.7-all.zip
```
Comment (quiqua) [inline gradle/wrapper/gradle-wrapper.properties]: Meh new gradle version available since like one hour ago: 
https://github.com/gradle/gradle/releases

**Review by elaydis — COMMENTED** [review]
🤷‍♀️  no idea about this spek thing 😝

**Pull request merged**